internal/signerverifier/ssh/ssh.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+package ssh
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/hiddeco/sshsig"
+	"golang.org/x/crypto/ssh"
+)
+
+// FIXME: make this the interface for Keys in TUF metadata
+type Key struct {
+	KeyType string
+	KeyVal  KeyVal
+	Scheme  string
+	keyID   string
+}
+
+type KeyVal struct {
+	Public string
+}
+
+func (k *Key) Verify(ctx context.Context, data []byte, sig []byte) error {
+
+	pub, err := parseSSH2Body(k.KeyVal.Public)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh public key material: %v", err)
+	}
+
+	signature, err := sshsig.Unarmor(sig)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh signature: %v", err)
+	}
+
+	message := bytes.NewReader(data)
+
+	// ssh-keygen uses sha512 to sign with **any*** key
+	hash := sshsig.HashSHA512
+	if err = sshsig.Verify(message, signature, pub, hash, "gittuf"); err != nil {
+		return fmt.Errorf("failed to verify ssh signature: %v", err)
+	}
+
+	return nil
+}
+
+func (k *Key) KeyID() (string, error) {
+	return k.keyID, nil
+}
+
+// FIXME: required by dsse Verifier interface; consider fixing interface
+func (k *Key) Public() crypto.PublicKey {
+	sshKey, _ := parseSSH2Body(k.KeyVal.Public)
+	return sshKey.(ssh.CryptoPublicKey).CryptoPublicKey()
+}
+
+// dsse.Signer interface implementation for ssh key signing
+// Includes `Key` as single source of truth for signing scheme and keyid
+type Signer struct {
+	Key  *Key
+	Path string
+}
+
+func (s *Signer) Sign(ctx context.Context, data []byte) ([]byte, error) {
+	cmd := exec.Command("ssh-keygen", "-Y", "sign", "-n", "gittuf", "-f", s.Path)
+
+	cmd.Stdin = bytes.NewBuffer(data)
+
+	output, err := cmd.Output()
+	if err != nil {
+		// TODO: Handle signing error (exit 255) on world-readable key file
+		//    git doesn't store read permissions; maybe need to chmod in test
+		return nil, fmt.Errorf("failed to run command %v: %v", cmd, err)
+	}
+
+	return output, nil
+
+}
+
+func (s *Signer) KeyID() (string, error) {
+	return s.Key.keyID, nil
+}
+
+// Import Key from path using ssh-keygen
+// Path can be public or private encrypted or plain key, much like git's
+// user.signingKey config
+func Import(path string) (*Key, error) {
+	cmd := exec.Command("ssh-keygen", "-e", "-f", path)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to run command %v: %v", cmd, err)
+	}
+
+	sshPub, err := parseSSH2Key(string(output))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse SSH2 key: %v", err)
+	}
+
+	return &Key{
+		keyID:   ssh.FingerprintSHA256(sshPub),
```

Comment:
Do we keep the `sha256` prefix?

Reply:
IMO this is a few bytes more on the wire vs. less complex code.  I don't have strong feelings, but I lean towards keeping it.